File: vertx-elasticsearch/src/main/java/fr/myprysm/vertx/elasticsearch/action/support/ReplicationResponse.java
package fr.myprysm.vertx.elasticsearch.action.support;

import io.vertx.codegen.annotations.DataObject;
import io.vertx.core.json.JsonObject;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
@DataObject(generateConverter = true)
public class ReplicationResponse {

    private ShardInfo shardInfo;

    public ReplicationResponse(ReplicationResponse other) {
        shardInfo = other.shardInfo;
    }

    public ReplicationResponse(JsonObject json) {
        ReplicationResponseConverter.fromJson(json, this);
    }

    public JsonObject toJson() {
        JsonObject json = new JsonObject();
        ReplicationResponseConverter.toJson(this, json);
        return json;
    }

}

File: vertx-elasticsearch/src/main/kotlin/fr/myprysm/vertx/elasticsearch/kotlin/action/support/ReplicationResponse.kt
package fr.myprysm.vertx.elasticsearch.kotlin.action.support

import fr.myprysm.vertx.elasticsearch.action.support.ReplicationResponse
import fr.myprysm.vertx.elasticsearch.action.support.ShardInfo

fun ReplicationResponse(
  shardInfo: fr.myprysm.vertx.elasticsearch.action.support.ShardInfo? = null): ReplicationResponse = fr.myprysm.vertx.elasticsearch.action.support.ReplicationResponse().apply {

  if (shardInfo != null) {
    this.setShardInfo(shardInfo)
  }
}

